test/UnitTests/MSTest.Core.Unit.Tests/Attributes/DynamicDataAttributeTests.cs

```diff
@@ -121,6 +121,41 @@ public void GetDisplayNameShouldReturnDisplayName()
             Assert.AreEqual("TestMethod1 (1,2,3)", displayName);
         }
 
+        [TestFrameworkV1.TestMethod]
+        public void GetDisplayNameShouldReturnDisplayNameWithDataRowDisplayName()
+        {
+            var data = new object[] { 1, 2, 3 };
+
+            this.dynamicDataAttribute.DynamicDisplayName = "GetCustomDynamicDataDisplayName";
+            var displayName = this.dynamicDataAttribute.GetDisplayName(this.testMethodInfo, data);
+            Assert.AreEqual("DynamicDataTestWithDisplayName TestMethod1 with 3 parameters", displayName);
+        }
+
+        [TestFrameworkV1.TestMethod]
+        public void GetDisplayNameShouldReturnDisplayNameWithDataRowDisplayNameInDifferentClass()
+        {
+            var data = new object[] { 1, 2, 3 };
+
+            this.dynamicDataAttribute.DynamicDisplayName = "GetCustomDynamicDataDisplayName2";
+            this.dynamicDataAttribute.DynamicDisplayNameDeclaringType = typeof(DummyTestClass2);
+            var displayName = this.dynamicDataAttribute.GetDisplayName(this.testMethodInfo, data);
+            Assert.AreEqual("DynamicDataTestWithDisplayName TestMethod1 with 3 parameters", displayName);
+        }
+
+        [TestFrameworkV1.TestMethod]
+        public void GetDisplayNameShouldThrowExceptionIfMethodDoesNotMatchExpectedSignature()
```

Comment:
Please add Tests for cases like : 
1. Method not existing i.e InvalidCustomDynamicDataDisplayName is not present in class
2. Method has wrong signature i.e. when first parameter doesn't match, second parameter doesn't match, return type doesn't match. Write tests for each of them separately.

Reply:
Done. Also picked up a bug where I wasn't checking that the method was static.

Reply:
Awesome 👍